postgres/datadog_checks/postgres/postgres.py

```diff
@@ -233,11 +240,18 @@ def dynamic_queries(self):
 
     def cancel(self):
         """
-        Cancels and waits for all threads to stop.
+        Cancels and waits for all threads to stop, and then
+        closes any open db connections
         """
-        self.statement_samples.cancel()
-        self.statement_metrics.cancel()
-        self.metadata_samples.cancel()
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            tasks = [
+                executor.submit(thread.cancel)
+                for thread in [self.statement_samples, self.statement_metrics, self.metadata_samples]
+            ]
+            concurrent.futures.wait(tasks)
```

Comment:
Should there be a timeout to this where we just continue forcefully? I'm thinking of some of the reports of the agent service not being able to be restarted because of the running hypothesis that one of our db integration wasn't able to gracefully terminate.

Reply:
yeah, i thought about adding that, better to be safe

Reply:
> I'm thinking of some of the reports of the agent service not being able to be restarted because of the running hypothesis that one of our db integration wasn't able to gracefully terminate

oh interesting, I hadn't heard about this